lib/paths-cache.js

```diff
@@ -140,20 +141,48 @@ export default class PathsCache extends EventEmitter {
     this._repositories = repositories.filter((r) => r !== null) // filter out non-repository directories
   }
 
+  /**
+   * Add watchers for all the projectDirectories
+   * @return {Promise<Array<Array<string>>}
+   * @private
+   */
+  async _addWatchers() {
+    const result = await Promise.all(
+      this._projectDirectories.map((projectDirectory) => this._addWatcherForDirectory(projectDirectory))
+    )
+    return result
+  }
+
+  /**
+   * Add a watcher for the projectDirectory
+   * @param  {Directory} projectDirectory
+   * @private
+   */
+  async _addWatcherForDirectory(projectDirectory) {
```

Comment:
I'm not sure why, but Atom's path watcher doesn't seem to work

Reply:
I tested on Atom stable

Reply:
Might be related to https://github.com/atom-community/autocomplete-paths/pull/262

Reply:
Also, it seems that `directory.onDidChange` is not recursive!

https://github.com/atom/node-pathwatcher#pathwatcherwatchfilename-listener

Reply:
We should switch to Chokidar:
https://www.npmjs.com/package/chokidar